Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license()" for more information.
import numpy as np
np.ndarray
<class 'numpy.ndarray'>
np.ndarray([1,2,3])_
SyntaxError: incomplete input
SyntaxError: incomplete input
SyntaxError: incomplete input
np.ndarray([1,2,3])
array([[[0., 0., 0.],
        [0., 0., 0.]]])
a=np.array([[1,2,3]])([1,2,3,4])
Traceback (most recent call last):
  File "<pyshell#5>", line 1, in <module>
    a=np.array([[1,2,3]])([1,2,3,4])
TypeError: 'numpy.ndarray' object is not callable
a=np.ndarray([[1,2,3]])([1,2,3,4])
Traceback (most recent call last):
  File "<pyshell#6>", line 1, in <module>
    a=np.ndarray([[1,2,3]])([1,2,3,4])
TypeError: 'list' object cannot be interpreted as an integer
a=np.array([[1,2,3],[4,5,6]])
a.shape
(2, 3)
a.size
6
a.ndin
Traceback (most recent call last):
  File "<pyshell#10>", line 1, in <module>
    a.ndin
AttributeError: 'numpy.ndarray' object has no attribute 'ndin'. Did you mean: 'ndim'?
a.ndim
2
print(a)
[[1 2 3]
 [4 5 6]]
a.squeeze
<built-in method squeeze of numpy.ndarray object at 0x000002115974FBD0>
np.expand_dims
<function expand_dims at 0x0000021159624550>
np.expand_dima()
Traceback (most recent call last):
  File "<pyshell#15>", line 1, in <module>
    np.expand_dima()
  File "C:\Users\Admin\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\numpy\__init__.py", line 284, in __getattr__
    raise AttributeError("module {!r} has no attribute "
AttributeError: module 'numpy' has no attribute 'expand_dima'. Did you mean: 'expand_dims'?








c=np.array([[1,2,3,],[3,4,5]])
                         
c=np.array([[1,2,3,],[3,4,5]])
                         
a.flatten()
                         
array([1, 2, 3, 4, 5, 6])
np.arange
                         
<built-in function arange>
a.transpose()
                         
array([[1, 4],
       [2, 5],
       [3, 6]])
np.arange(45)
                         
array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
       17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33,
       34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44])
np.arange(45).reshape[5,1,9]
                         
Traceback (most recent call last):
  File "<pyshell#28>", line 1, in <module>
    np.arange(45).reshape[5,1,9]
TypeError: 'builtin_function_or_method' object is not subscriptable
np.arange(45).reshape(5,1,9)
                         
array([[[ 0,  1,  2,  3,  4,  5,  6,  7,  8]],

       [[ 9, 10, 11, 12, 13, 14, 15, 16, 17]],

       [[18, 19, 20, 21, 22, 23, 24, 25, 26]],

       [[27, 28, 29, 30, 31, 32, 33, 34, 35]],

       [[36, 37, 38, 39, 40, 41, 42, 43, 44]]])
np.arange(45).reshape(15,1,3)
                         
array([[[ 0,  1,  2]],

       [[ 3,  4,  5]],

       [[ 6,  7,  8]],

       [[ 9, 10, 11]],

       [[12, 13, 14]],

       [[15, 16, 17]],

       [[18, 19, 20]],

       [[21, 22, 23]],

       [[24, 25, 26]],

       [[27, 28, 29]],

       [[30, 31, 32]],

       [[33, 34, 35]],

       [[36, 37, 38]],

       [[39, 40, 41]],

       [[42, 43, 44]]])
np.arange(45).reshape(5,1,3)
                         
Traceback (most recent call last):
  File "<pyshell#31>", line 1, in <module>
    np.arange(45).reshape(5,1,3)
ValueError: cannot reshape array of size 45 into shape (5,1,3)

np.arange(45).reshape(45,1,1)
                         
array([[[ 0]],

       [[ 1]],

       [[ 2]],

       [[ 3]],

       [[ 4]],

       [[ 5]],

       [[ 6]],

       [[ 7]],

       [[ 8]],

       [[ 9]],

       [[10]],

       [[11]],

       [[12]],

       [[13]],

       [[14]],

       [[15]],

       [[16]],

       [[17]],

       [[18]],

       [[19]],

       [[20]],

       [[21]],

       [[22]],

       [[23]],

       [[24]],

       [[25]],

       [[26]],

       [[27]],

       [[28]],

       [[29]],

       [[30]],

       [[31]],

       [[32]],

       [[33]],

       [[34]],

       [[35]],

       [[36]],

       [[37]],

       [[38]],

       [[39]],

       [[40]],

       [[41]],

       [[42]],

       [[43]],

       [[44]]])

np.arange(45).reshape(3,5,3)
                         
array([[[ 0,  1,  2],
        [ 3,  4,  5],
        [ 6,  7,  8],
        [ 9, 10, 11],
        [12, 13, 14]],

       [[15, 16, 17],
        [18, 19, 20],
        [21, 22, 23],
        [24, 25, 26],
        [27, 28, 29]],

       [[30, 31, 32],
        [33, 34, 35],
        [36, 37, 38],
        [39, 40, 41],
        [42, 43, 44]]])
a[0,1]
                         
2
3
                         
3
a[01,0,0]
                         
SyntaxError: leading zeros in decimal integer literals are not permitted; use an 0o prefix for octal integers
a[0,1,1]
                         
Traceback (most recent call last):
  File "<pyshell#39>", line 1, in <module>
    a[0,1,1]
IndexError: too many indices for array: array is 2-dimensional, but 3 were indexed
a[1,0,0,]
                         
Traceback (most recent call last):
  File "<pyshell#40>", line 1, in <module>
    a[1,0,0,]
IndexError: too many indices for array: array is 2-dimensional, but 3 were indexed
a[1,0,0]
                         
Traceback (most recent call last):
  File "<pyshell#41>", line 1, in <module>
    a[1,0,0]
IndexError: too many indices for array: array is 2-dimensional, but 3 were indexed


np.dtype='float'
                         
np.stack(x,y)